api/app.rb

```diff
@@ -236,30 +217,39 @@ class RenderApp < Sinatra::Base
 
   # TODO: The :id should be parsed against the same regex as above
   post '/:id' do
-    template = Template.new(id: params['id'])
-    if template.valid?
-      response.headers['Content-Type'] = 'text/plain'
-      script = Script.new(template: template, user: @current_user)
-
-      # This conditional should not be reached ATM
-      unless script.valid?
-        status 500
-        halt
-      end
+    # NOTE: The template is only loaded so it can preform the CSV conversion for
+    # multiselect questions. Removing support for CSV would drop this requirement
+    template = Template.find(params['id'], user: @current_user)
+    if template.nil?
+      status 404
+      halt
+    end
 
-      begin
-        script.render_and_save(**params.to_h.transform_keys(&:to_sym))
-      rescue
-        FlightJobScriptAPI.logger.debug("Rendering script failed") { $! }
-        status 422
-        halt
-      end
+    # Transform any multiselect answers from CSV to Arrays
+    answers = params.to_h.dup
+    template.generation_questions.each do |q|
+      next unless q.metadata['format']['type'] == 'multiselect'
+      next unless answers[q.id].is_a? String
+      FlightJobScriptAPI.logger.warn "Implicitly converting '#{q.id}' from CSV to an array"
+      answers[q.id] = CSV.parse(answers[q.id]).first
+    end
```

Comment:
Yep, the CSV translation exists because of how the client operates. Initially when I implemented the CLI, I didn't realise it was in the CSV format as opposed to atomic values. So the CLI when down the path of arrays, which in many ways is easier.

However this has left us with a disconnect between arrays in the CLI and CSV in the web client. There is another translation in the serialisation's of the the questions' defaults as document here:
https://github.com/openflighthpc/flight-job-script-service/blob/dev/active-jobs/api/docs/routes.md#get---templatesidquestions

If you would prefer the client worked with `arrays`, then all the CSV handling could be removed.

Reply:
I think using JSON arrays would be simpler.  Can you make the changes to the API and I'll update the client.

Reply:
1930e1df42c47c8ce1354601b22f497800aab7e0